dev/tools/gen_defaults/lib/search_bar_template.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'template.dart';
+
+class SearchBarTemplate extends TokenTemplate {
+  const SearchBarTemplate(super.blockName, super.fileName, super.tokens, {
+    super.colorSchemePrefix = '_colors.',
+    super.textThemePrefix = '_textTheme.'
+  });
+
+  String _textStyle(String token) {
+    if (tokens.containsKey('$token.text-style')) {
+      return '''
+
+    MaterialStatePropertyAll<TextStyle?>(${textStyle(token)}?.copyWith(color: ${componentColor(token)}))''';
+    }
+    return '''null''';
+  }
+
+  @override
+  String generate() => '''
+class _SearchBarDefaultsM3 extends SearchBarThemeData {
+  _SearchBarDefaultsM3(this.context);
+
+  final BuildContext context;
+  late final ColorScheme _colors = Theme.of(context).colorScheme;
+  late final TextTheme _textTheme = Theme.of(context).textTheme;
+
+  @override
+  MaterialStateProperty<Color?>? get backgroundColor =>
+    MaterialStatePropertyAll<Color>(${componentColor("md.comp.search-bar.container")});
+
+  @override
+  MaterialStateProperty<double>? get elevation =>
+    const MaterialStatePropertyAll<double>(${elevation("md.comp.search-bar.container")});
+
+  @override
+  MaterialStateProperty<Color>? get shadowColor =>
+    MaterialStatePropertyAll<Color>(_colors.shadow);
+
+  @override
+  MaterialStateProperty<Color>? get surfaceTintColor =>
+    MaterialStatePropertyAll<Color>(${colorOrTransparent("md.comp.search-bar.container.surface-tint-layer.color")});
+
+  @override
+  MaterialStateProperty<Color?>? get overlayColor =>
+    MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.focused)) {
```

Comment:
When pressed is true, hovered is also typically true, so we should tests like this should be [disabled], pressed, hovered, focused.